docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,282 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high-quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in `CLIP-as-service` support indexing a very large dataset (millions/billions) and querying within 50ms, depending on the machine.
+
+In order to implement retrieval, we add an [`AnnLite`](https://github.com/jina-ai/annlite) indexer executor after the encoder executor in CLIP-as-service.
+
+
```

Comment:
a table of contents would be nice

Reply:
a table about what?

Reply:
[something like this](https://en.wikipedia.org/wiki/Table_of_contents) :)

Reply:
I think we already have one here on the right side of the page: 
<img width="286" alt="Screen Shot 2022-09-07 at 2 47 32 PM" src="https://user-images.githubusercontent.com/16580382/188808394-1d701b4d-8713-4634-9686-30c6750ab766.png">